rust/parquet/src/arrow/arrow_writer.rs

```diff
@@ -176,19 +176,175 @@ fn write_leaves(
             }
             Ok(())
         }
+        ArrowDataType::Dictionary(key_type, value_type) => {
+            use arrow_array::{
+                Int16DictionaryArray, Int32DictionaryArray, Int64DictionaryArray,
+                Int8DictionaryArray, PrimitiveArray, StringArray, UInt16DictionaryArray,
+                UInt32DictionaryArray, UInt64DictionaryArray, UInt8DictionaryArray,
+            };
+            use ArrowDataType::*;
+            use ColumnWriter::*;
+
+            let array = &**array;
+            let mut col_writer = get_col_writer(&mut row_group_writer)?;
+            let levels = levels.pop().expect("Levels exhausted");
+
+            macro_rules! dispatch_dictionary {
+                ($($kt: pat, $vt: pat, $w: ident => $kat: ty, $vat: ty,)*) => (
+                    match (&**key_type, &**value_type, &mut col_writer) {
+                        $(($kt, $vt, $w(writer)) => write_dict::<$kat, $vat, _>(array, writer, levels),)*
+                        (kt, vt, _) => panic!("Don't know how to write dictionary of <{:?}, {:?}>", kt, vt),
```

Comment:
@carols10cents do we have to worry about this panic? If so, we can still address it in a future PR. I must have missed it over the weekend.

Reply:
I think this should probably be `unreachable!`; it's similar to [this spot](https://github.com/apache/arrow/pull/8402/commits/91a22bb00a202e10f3f5fc2970bea726baddfb03) where the code shouldn't get to that spot.